src/main/kotlin/org/rust/ide/inspections/UnnecessaryParenthesisInspection.kt

```diff
@@ -0,0 +1,67 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFixBase
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustBinaryExpr
+import org.rust.lang.core.psi.RustIfExpr
+import org.rust.lang.core.psi.RustParenExpr
+import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustParenExprImpl
+import org.rust.lang.core.psi.util.parentRelativeRange
+
+class UnnecessaryParenthesisInspection : RustLocalInspectionTool() {
+    override fun getDisplayName(): String = "Unnecessary parenthesis"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : RustVisitor() {
+            override fun visitParenExpr(expr: RustParenExpr) {
+                val subExpr = expr.expr
+                when (subExpr) {
+                    !is RustBinaryExpr -> {
+                        if (expr.parent !is RustIfExpr)
+                            holder.registerProblem(expr, "Redundant parenthesis around unambiguous expression", RemoveParenthesis)
```

Comment:
I am not quite sure that this is entirely correct. Let's concentrate on the most important problem of redundant parenthesis in `if`. 

If we got it working, adding more cases would be easy.